addons/koji/common/src/main/java/org/commonjava/indy/koji/content/KojiContentManagerDecorator.java

```diff
@@ -350,7 +411,48 @@ public Transfer getTransfer( final ArtifactStore store, final String path, final
         }
     }
 
-    private RemoteRepository createRemoteRepository(ArtifactRef artifactRef, final KojiBuildInfo build, final KojiSessionInfo session)
+    private ArtifactStore getAuthStore(){
+        if ( config.getBuildAuthorityStore() != null )
+        {
+            final StoreKey authStoreKey = StoreKey.fromString( config.getBuildAuthorityStore() );
+            try
+            {
+                return storeDataManager.getArtifactStore( authStoreKey );
+            }
+            catch ( IndyDataException e )
+            {
+                Logger logger = LoggerFactory.getLogger( getClass() );
+                logger.warn( "Error occurred when finding authoritative store for {}, error message: {}",
+                             authStoreKey, e.getMessage() );
+            }
+        }
+        return null;
+    }
+
+    private String checksumArtifact( ArtifactStore store, String path )
+    {
+        final Logger logger = LoggerFactory.getLogger( getClass() );
+        try
+        {
+            if ( delegate.exists( store, path ) )
+            {
+                final ArtifactData artifactData = delegate.digest( store.getKey(), path, ContentDigest.MD5 );
```

Comment:
This triggers download of the binary artifact and computation of the checksum from the downloaded data, but I suppose we can rely on the hosted checksums in the authoritative repo and perform this action only if the checksum is not hosted for the artifact. On the other hand the best way how to achieve that is probably to do it directly in the digest method by adding boolean parameter like useHostedChecksums to the method with default false and maybe it is out of scope of this change.